src/actions/pause/mapFrames.js

```diff
@@ -55,6 +64,63 @@ export function mapDisplayNames(
   });
 }
 
+function isWasmOriginalSourceFrame(frame, getState: () => State): boolean {
+  if (isGeneratedId(frame.location.sourceId)) {
+    return false;
+  }
+  const generatedSource = getSourceFromId(
+    getState(),
+    frame.generatedLocation.sourceId
+  );
+  return generatedSource.isWasm;
+}
+
+async function expandFrames(
+  frames: Frame[],
+  getState: () => State
+): Promise<Frame[]> {
+  const result = [];
+  for (let i = 0; i < frames.length; ++i) {
+    const frame = frames[i];
+    if (
+      frame.isOriginalSourceFrame ||
+      !isWasmOriginalSourceFrame(frame, getState)
+    ) {
+      result.push(frame);
+      continue;
+    }
+    const originalFrames = await getOriginalStackFrames(
+      frame.generatedLocation
+    );
+    if (!originalFrames) {
+      result.push(frame);
+      continue;
+    }
+
+    assert(originalFrames.length > 0, "Expected at least one original frame");
+    originalFrames[0].originalLocation = frame.location;
```

Comment:
what is `originalLocation` used for? it looks like a new field

Reply:
that's mistake (last minute refactoring) -- fixed and will add a comment